elisa/src/client/RunDetails/components/CalibrationCurvePanel.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { PureComponent } from 'react';
+import $ from 'jquery';
+import { getServerContext } from '@labkey/api';
+import { generateId } from '@labkey/components';
+
+import { getPlotConfigFromOptions } from "../utils";
+import { CurveFitData, PlotOptions } from "../models";
+import { PlotButtonBar } from "./PlotButtonBar";
+
+interface Props {
+    protocolId: number,
+    runId: number,
+    runPropertiesRow: {[key: string]: any},
+    data: any[],
+    plotOptions: PlotOptions,
+    curveFitData: CurveFitData,
+    columnInfo: {[key: string]: any}
+}
+
+interface State {
+    plotId: string
+}
+
+export class CalibrationCurvePanel extends PureComponent<Props, State> {
+    state: Readonly<State> = {
+        plotId: generateId('calibration-curve-')
+    };
+
+    componentDidMount(): void {
+        this.renderPlot();
+    }
+
+    componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>, snapshot?: any): void {
+        this.renderPlot();
+    }
+
+    getPlotElement() {
+        return $('#' + this.state.plotId);
+    }
+
+    getPlotWidth(): number {
+        return $('.plot-panel-section').width();
+    }
```

Comment:
Wonder if you could use React Refs and not need the dependency on jQuery

Reply:
Yes, done.